components/brave_wallet/browser/android_page_appearing_browsertest.cc

```diff
@@ -436,4 +436,36 @@ IN_PROC_BROWSER_TEST_F(AndroidPageAppearingBrowserTest, TestSendPageAppearing) {
       console_observer, blink::mojom::ConsoleMessageLevel::kWarning,
       {"TypeError: Cannot read properties of undefined (reading 'forEach')"});
 }
+
+IN_PROC_BROWSER_TEST_F(AndroidPageAppearingBrowserTest,
+                       TestDepositPageAppearing) {
```

Comment:
These three test suites have lots of redundant code basically the only difference is url, they should be run in a loop

Reply:
Do you mean a single test that runs them in a loop? Can you link an example I can use for reference, and I'll do mt best to replicate 😁

Reply:
Yes, just needs a single test suite called "PagesAppearing"
Here is one of the example
https://github.com/brave/brave-core/blob/14f77cbbd12e57e8aec871657ff5b40bec941fe0/browser/brave_wallet/solana_provider_renderer_browsertest.cc#L1221

Reply:
if you don't need multiple fields, then just loop through urls like this
```
for (const GURL& url : {GURL("chrome://wallet/swap"), 
                        GURL("chrome://wallet/send"),
                        GURL("chrome://wallet/crypto/deposit-funds")}) {
  content::NavigationController::LoadURLParams params(url);
...
}
```

Reply:
Thank you for pointing me in the right direction @darkdh .
I tried something like this:

```c++
IN_PROC_BROWSER_TEST_F(AndroidPageAppearingBrowserTest, PagesAppearing) {
  for (const GURL& url : {GURL("chrome://wallet/swap"), 
                        GURL("chrome://wallet/send"),
                        GURL("chrome://wallet/crypto/deposit-funds")}) {
    content::NavigationController::LoadURLParams params(url);
    params.transition_type = ui::PageTransitionFromInt(
        ui::PAGE_TRANSITION_TYPED | ui::PAGE_TRANSITION_FROM_ADDRESS_BAR);

    auto* web_contents = GetActiveWebContents();
    ...
```

But it gives the error: `[ RUN      ] AndroidPageAppearingBrowserTest.PagesAppearing
[FATAL:browser_test_base.cc(256)] Check failed: !g_instance_already_created. Each browser test should be run in a new process. If you are adding a new browser test suite that runs on Android, please add it to //build/android/pylib/gtest/gtest_test_instance.py.`

![Screenshot from 2023-07-19 11-56-39](https://github.com/brave/brave-core/assets/3308503/88495114-d628-441f-bdb9-81e54fbe6b60)

Do you have any suggestions on how to solve this? Thanks.

Reply:
ok, I don't know what is going on here. Changing it to loop should not make ctor of BrowserTestBase to be called multiple times. cc @SergeyZhukovsky 
Let's do it another way, just put those code into member function of AndroidPageAppearingBrowserTest and be shared between three test suites and each suite calls that function with its own url.

Reply:
Moved the code into a member function of `AndroidPageAppearingBrowserTest` but now the third test fails with a different error message. I'm a bit confused here.
 I'll ping @SergeyZhukovsky and see if we can find a solution.

![image](https://github.com/brave/brave-core/assets/3308503/bb6714d3-4bc7-45ec-85cc-1195829a255a)

Reply:
🎉 Solution found! 
Code duplication removed in [`6677290` (#19256)](https://github.com/brave/brave-core/pull/19256/commits/6677290059c93144c2c46a7402676cfd4db73f1d)